connection-cache/src/connection_cache.rs

```diff
@@ -18,38 +19,49 @@ use {
 // Should be non-zero
 pub static MAX_CONNECTIONS: usize = 1024;
 
-/// Used to decide whether the TPU and underlying connection cache should use
-/// QUIC connections.
-pub const DEFAULT_TPU_USE_QUIC: bool = true;
+/// Default connection pool size per remote address
+pub const DEFAULT_CONNECTION_POOL_SIZE: usize = 4;
 
-/// Default TPU connection pool size per remote address
-pub const DEFAULT_TPU_CONNECTION_POOL_SIZE: usize = 4;
-
-pub const DEFAULT_TPU_ENABLE_UDP: bool = false;
+pub trait ConnectionManager: Sync + Send {
+    fn new_connection_pool(&self) -> Box<dyn ConnectionPool>;
+    fn new_connection_config(&self) -> Box<dyn NewConnectionConfig>;
+    fn get_port_offset(&self) -> u16;
+}
 
-pub struct TpuConnectionCache<P: ConnectionPool> {
-    pub map: RwLock<IndexMap<SocketAddr, P>>,
+pub struct ConnectionCache {
```

Comment:
This is a pity, though. The generic was an attempt to avoid dynamic dispatch.
I suppose we're probably not ready for that yet, since we're moving other services to quic and would like to slowly transition.